src/index.rs

```diff
@@ -724,48 +787,44 @@ impl Index {
             .insert(qualified_name.to_lowercase(), entry);
     }
 
-    pub fn register_global_variable(
-        &mut self,
+    pub fn create_global_variable(
```

Comment:
so this no longer registers? will this be confusing? it feels unique for global variables

Reply:
I moved the register to its own call that takes a created variable because of the builder thing/ so that clippy is happy. I'll take another look I also felt it wasn't always needed.

Reply:
can we move this method into VariableIndexEntry maybe?

Reply:
I deleted the methods, I think they are redundant when using the new builder